cibuildwheel/docker_container.py

```diff
@@ -88,12 +199,24 @@ def __exit__(
     ) -> None:
 
         self.bash_stdin.close()
+
+        if self._sleep_time:
+            time.sleep(self._sleep_time)
+
         self.process.terminate()
         self.process.wait()
 
+        # When using podman there seems to be some race condition. Give it a
+        # bit of extra time.
+        if self._sleep_time:
+            time.sleep(self._sleep_time)
+
         assert isinstance(self.name, str)
 
-        subprocess.run(["docker", "rm", "--force", "-v", self.name], stdout=subprocess.DEVNULL)
+        subprocess.run(
+            [self.oci_exe, "rm"] + self._common_args + ["--force", "-v", self.name],
+            stdout=subprocess.DEVNULL,
+        )
         self.name = None
```

Comment:
@Erotemic The race condition was happening here you said? I tried:
```python
self.bash_stdin.close()

assert isinstance(self.name, str)
subprocess.run(["docker", "rm", "--force", "-v", self.name], stdout=subprocess.DEVNULL)
self.process.terminate()
self.process.wait()
self.name = None
```

[I copied your `cwd` changes and along with this](https://github.com/jshowacre/cibuildwheel/commit/2e51c6e44b7a429048bb42f06bb9839e4bdbf021), the unit tests and sample project seemed to work! Could you try as well?

Reply:
It still seems to break for me without the sleeps. The error I get is:

```
Error: read unixpacket @->/proc/self/fd/15/attach: read: connection reset by peer

                                                              ✕ 1.09s
Error: Command podman exec --cgroup-manager=cgroupfs --storage-driver=vfs --root=/root/.local/share/containers/vfs-storage/ -i cibuildwheel-19e7f1f5-ddd7-48f5-a130-7dc62c9776d1 tar -cC /output -f /tmp/output-cibuildwheel-19e7f1f5-ddd7-48f5-a130-7dc62c9776d1.tar . failed with code 255. None
```

Note, this is with the vfs and cgroupfs options:

```python
          export CIBW_SKIP="pp* cp310-* cp39-* cp37-* cp36-* *musllinux*"
          export CIBW_OCI_EXE="podman"
          export CIBW_OCI_EXTRA_ARGS_CREATE="--events-backend=file --privileged"
          export CIBW_OCI_EXTRA_ARGS_COMMON="--cgroup-manager=cgroupfs --storage-driver=vfs --root=$HOME/.local/share/containers/vfs-storage/"
          export CIBW_OCI_EXTRA_ARGS_START="--events-backend=file --cgroup-manager=cgroupfs --storage-driver=vfs"
          cibuildwheel --output-dir wheelhouse --platform linux --archs x86_64
```

When I add them back in it goes back to working.

I further verified this by adding an oci-sleep-time option, so within the same test session I set the variable to `export CIBW_OCI_SLEEP_TIME="0.0"` and I again got an error:

```
ERRO[0000] container not running                        
ERRO[0022] Error forwarding signal 15 to container 56aa55d43bfc75cf39d65ca4a86037ddeb129f3fe7d780c6b3de8604b160d9f1: error sending signal to container 56aa55d43bfc75cf39d65ca4a86037ddeb129f3fe7d780c6b3de8604b160d9f1: `/usr/sbin/runc kill 56aa55d43bfc75cf39d65ca4a86037ddeb129f3fe7d780c6b3de8604b160d9f1 15` failed: exit status 1 
```

and then `export CIBW_OCI_SLEEP_TIME="0.1"`, and again it worked.